bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/EvccHandler.java

```diff
@@ -129,27 +142,41 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                         } else if (command == OnOffType.OFF) {
                             evccAPI.unsetTargetCharge(loadpoint);
                             targetTimeEnabled = false;
+                        } else {
+                            throw new EvccException("Command has wrong type, OnOffType required!");
                         }
                         break;
                     case CHANNEL_LOADPOINT_PHASES:
                         if (command instanceof DecimalType) {
                             evccAPI.setPhases(loadpoint, ((DecimalType) command).intValue());
+                        } else {
+                            throw new EvccException("Command has wrong type, DecimalType required!");
                         }
                         break;
                     case CHANNEL_LOADPOINT_MIN_CURRENT:
                         if (command instanceof QuantityType) {
-                            evccAPI.setMinCurrent(loadpoint, ((QuantityType<?>) command).intValue());
+                            evccAPI.setMinCurrent(loadpoint,
+                                    ((QuantityType<?>) command).toUnit(Units.AMPERE).intValue());
+                        } else if (command instanceof DecimalType) {
+                            evccAPI.setMinCurrent(loadpoint, ((DecimalType) command).intValue());
+                        } else {
+                            throw new EvccException("Command has wrong type, QuantityType or DecimalType required!");
                         }
                         break;
                     case CHANNEL_LOADPOINT_MAX_CURRENT:
                         if (command instanceof QuantityType) {
-                            evccAPI.setMaxCurrent(loadpoint, ((QuantityType<?>) command).intValue());
+                            evccAPI.setMaxCurrent(loadpoint,
+                                    ((QuantityType<?>) command).toUnit(Units.AMPERE).intValue());
+                        } else if (command instanceof DecimalType) {
+                            evccAPI.setMaxCurrent(loadpoint, ((DecimalType) command).intValue());
+                        } else {
+                            throw new EvccException("Command has wrong type, QuantityType or DecimalType required!");
                         }
                         break;
                     default:
                         return;
                 }
-            } catch (EvccApiException e) {
+            } catch (EvccException e) {
```

Comment:
Should `refresh()` below be called when command is invalid?

Reply:
Yes, because otherwise the Item would have a wrong state.
I don't know why the Item is auto updating on command, because I set the autoUpdatePolicy for the read-write channels to veto, but it does.